src/api/app.js

```diff
@@ -12,6 +24,17 @@ export default {
     }
     xhr.send()
   },
+  getGithubUsername: function (email, callback) {
+    const url = `https://api.github.com/search/commits?q=author-email:${email}&sort=author-date&per_page=1`
+    const headers = [{
+      key: 'Accept',
+      value: 'application/vnd.github.cloak-preview'
+      // TODO add token
```

Comment:
It needs a token?

Reply:
We do not "need" one but good if we can add one.

One funny thing I noticed is that when the user is authenticated via OAuth2, we have the Github token that we can use for each user to authenticate the request to fetch Github username but sadly we only have that token when the user signs in, once the user leaves the app and come back, firebase does not return the token in the [onAuthStateChanged](https://github.com/samiheikki/javascript-guessing-game/pull/282/files#diff-ea994a51b0d9884f4fc046eca11a5c99R80) method; in fact, it does not persist it at all. So we probably have to persist the token ourselves when the user signs in.

But personally, I am not sure there is much need for that much logic. I think this works fine for the app for now